### PR TITLE
Feature - Add Vagrant Support

### PR DESCRIPTION
**Pull request by joshuacherry**
I've added a Vagrantfile which uses an Ubuntu14.04 box matching the same `docker-engine` and `docker-compose` versions used from `install-docker.sh` for Travis tests.

This should offer a standard platform between any contributors interested in testing with the same version of docker that Travis is using for tests. I have on more than one occasion had tests pass locally only to fail in Travis before a PR because the docker-engine version used in Travis had some weird compatibility issues with certain Ansible versions.

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+$setupScript = <<SCRIPT
+echo provisioning docker...
+sudo apt-get update
+sudo apt-key adv --keyserver hkp://p80.pool.sks-keyservers.net:80 --recv-keys 58118E89F3A912897C070ADBF76221572C52609D
+sudo apt-add-repository 'deb https://apt.dockerproject.org/repo ubuntu-trusty main'
+sudo apt-get update
+sudo apt-get -o Dpkg::Options::="--force-confnew" install --force-yes -y docker-engine=1.10.2-0~trusty
+sudo usermod -a -G docker vagrant
+curl -L "https://github.com/docker/compose/releases/download/1.6.2/docker-compose-$(uname -s)-$(uname -m)" > /usr/local/bin/docker-compose
```
Comment (benwebber) [inline Vagrantfile]: Please validate the SHA256 sum after downloading.

File: Vagrantfile [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+chmod +x /usr/local/bin/docker-compose
+docker version
+
+docker-compose version
+SCRIPT
+
+Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
+  config.vm.box = "bento/ubuntu-14.04"
```
Comment (benwebber) [inline Vagrantfile:26]: Why not the official Ubuntu box?

File: Vagrantfile [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+  my_machines.each do |short_name, array|
```
Comment (benwebber) [inline Vagrantfile]: This can be removed; let Vagrant assign an IP automatically.

File: README.rst [hunk 4]
```diff
@@ -205,16 +205,23 @@ If you'd like to help with the project itself, here are some other ways you can
...
+This role includes a Vagrantfile used with a Docker-based test harness for integration testing. Using Vagrant allows all contributors to test on the same platform and avoid false test failures due to untested or incompatible docker versions.
 
-This role includes a Docker-based test harness for integration testing.
+1. Install `Vagrant <https://www.vagrantup.com/>`__ and `VirtualBox <https://www.virtualbox.org/>`__.
 
-1. Install `Docker <https://docs.docker.com/engine/installation/>`__ and `Docker Compose <https://docs.docker.com/compose/>`__.
+2. Run ``vagrant up`` from the same directory as the Vagrantfile in this repository.
 
-2. Run tests with ``make``.
+3. SSH into the VM with: ``vagrant ssh`` or ``ssh 127.0.0.1:2222`` or ``ssh 10.1.15.10``
```
Comment (benwebber) [inline README.rst]: This should work: `vagrant ssh -c "make -C /vagrant <target>"`.

You can also wrap that as a script called something like `vagrant-make`.

File: README.rst [hunk 5]
```diff
@@ -205,16 +205,23 @@ If you'd like to help with the project itself, here are some other ways you can
...
+
+4. Change directories into **/vagrant** with the command: ``cd /vagrant``
+
+5. Test the role against each Dockerfile with ``make``:
 
    ::
 
        make jessie64 test
```
Comment (benwebber) [inline README.rst]: We can suggest running `make help`, which will list all the available targets.

File: README.rst [hunk 6]
```diff
@@ -205,16 +205,23 @@ If you'd like to help with the project itself, here are some other ways you can
 
 Testing
 ~~~~~~~
+This role includes a Vagrantfile used with a Docker-based test harness for integration testing. Using Vagrant allows all contributors to test on the same platform and avoid false test failures due to untested or incompatible docker versions.
```
Comment (benwebber) [inline README.rst]: This role used to use Vagrant for testing, but I switched to Docker because it uses much fewer resources.

Using Vagrant would indeed help prevent a specific test failure mode, as you've pointed out in the README.

Could you rephrase this to imply using the Vagrant box is optional? Maybe something like "We also offer a Vagrant box that approximates the Travis CI setup. Here's how to use it:".

**Review by benwebber — CHANGES_REQUESTED** [review]

File: Vagrantfile (hunk 2 above)
Comment (joshuacherry) [reply Vagrantfile:26]: The official Ubuntu box stopped following Vagrant's requirement for ssh users starting in their Ubuntu 16 box,  https://bugs.launchpad.net/cloud-images/+bug/1569237

The bento boxes are managed by Chef which I would consider relatively trustworthy.  
https://github.com/chef/bento

File: README.rst (hunk 4 above)
Comment (joshuacherry) [reply README.rst]: SInce we could have people developing from multiple different platforms, I would like to avoid using scripts that run on their local machine. For example a windows users cannot use `vagrant ssh` by default and instead will typically use programs like putty or SecureCRT to establish their ssh connection.

File: Vagrantfile (hunk 1 above)
Comment (joshuacherry) [reply Vagrantfile]: Then we should also consider validating the SHA256 sum in the `install-docker.sh` script.  
https://github.com/devops-coop/ansible-minecraft/blob/master/scripts/install-docker.sh#L5-L8  
  
Alternatively, I'll also look into using pip for this. Brief testing shows the below code has the same intended outcome:
```
$ sudo apt-get install python-pip
$ sudo pip install --upgrade pip
$ sudo pip install docker-compose==1.6.2
$ docker-compose version                                
docker-compose version 1.6.2, build 4d72027
```

File: Vagrantfile (hunk 2 above)
Comment (benwebber) [reply Vagrantfile:26]: That's absurd. Thanks.

**Review by benwebber — APPROVED** [review]
